priceloop_api/priceloop/test/test_table.py

```diff
@@ -0,0 +1,145 @@
+import os
+
+import pandas as pd
+import pytest
+
+from priceloop_api.priceloop.extras import (
+    Column,
+    ColumnMixedTypes,
+    ColumnType,
+    PatchMode,
+    Table,
+    TableUtils,
+    get_priceloop_client,
+)
+
+
+@pytest.fixture
+def sample_table() -> Table:
+    return Table(
+        os.environ["NOCODE_WORKSPACE"],
+        "table1",
+        [
+            Column("name", ColumnType.Text),
+            Column("age", ColumnType.Number),
+            Column("date_of_birth", ColumnType.Date),
+            Column("gpa", ColumnType.Number),
+            Column("json_field", ColumnType.Json),
+        ],
+    )
+
+
+@pytest.fixture
+def sample_dataframe1():
+    df = pd.DataFrame(
+        data={
+            "name": ["John", "Alice", "Bob"],
+            "age": [25, 30, 22],
+            "date_of_birth": pd.to_datetime(["1997-05-15", "1992-08-22", "2000-01-10"]),
+            "gpa": [3.3, 2.9, 3.0],
+            "json_field": ["{}", '{"a":"object"}', '{"1":3}'],
+        },
+    )
+    return df
+
+
+@pytest.fixture
+def sample_dataframe2():
+    return pd.DataFrame(
+        data={
+            "name": ["John", 22, "Bob"],
+            "age": [25, 30, 22],
+            "date_of_birth": pd.to_datetime(["1997-05-15", "1992-08-22", "2000-01-10"]),
+            "gpa": [3.3, 2.9, 3.0],
+            "json_field": ["{}", '{"a":"object"}', '{"1":3}'],
+        },
+    )
+
+
+class TestTable:
+    def test_set_df_success(self, sample_table, sample_dataframe1):
+        sample_table.set_df(sample_dataframe1)
+
+    def test_set_df_fail(self, sample_table, sample_dataframe2):
+        assert True
+        try:
+            sample_table.set_df(sample_dataframe2)
+            assert False
```

Comment:
do you want to let if fail? There is e.g. `pytest.fail("Failed for some reason")`